students/brgalloway/Lesson_6/test_mailroom_part_4.py

```diff
@@ -0,0 +1,71 @@
+from mailroom_part_4 import *
+import os
+
+a_dict = {
+        "Jeff Bezos": {
+            "donation_total": 877.33,
+            "times_donated": 1,
+            "average_donation": 877.33
+        },
+        "Paul Allen": {
+            "donation_total": 708.42,
+            "times_donated": 3,
+            "average_donation": 236.14
+        }
+    }
+
+'''
+ensure that list of names returned is a string
+'''
+def test_list_names():
+    names = list_names()
+    assert type(names) is str
+
+'''
+Verify sorted donors is always returned as a list
+'''
+def test_sort_donors():
+    output = sorted(a_dict.items(), key=sort_donors, reverse=True)
+    assert type(output) is list
+
+'''
+Test to ensure any dictionary passed into report is formatted properly
+and returns the correct values and type based on key 
+'''
+def test_generate_report():
+    single_report = generate_report(a_dict)
+    assert donors_list["Paul Allen"]["donation_total"] == 708.42
+    assert type(donors_list["Paul Allen"]["donation_total"]) == float
```

Comment:
no need to check for the type, the assert on line 37 already takes care of it. In general, in Python you would rarely check for type in tests.